python/rapidsmpf/rapidsmpf/integrations/core.py

```diff
@@ -82,6 +83,20 @@ def get_new_shuffle_id(get_occupied_ids: Callable[[], Sequence[set[int]]]) -> in
         return _shuffle_id_vacancy.pop()
 
 
+_STATISTICS_KEYS = [
```

Comment:
I'd like to avoid repeating statistic names here. Do we have a list of valid statistic names in C++ somewhere (looking at https://github.com/rapidsai/rapidsmpf/blob/d5aefa1d43f987115d7722b979a93414bcb3527f/cpp/src/shuffler/shuffler.cpp#L412, I think the answer is "no")?

Reply:
Yeah, I think the answer is no, they are added on spot when needed, so this is likely to change over time as well.

Reply:
What do people think about using an enum (in C++) and updating `Statistics.add_stat` to take the enum member rather than a string?

For consumers of these statistics, it's helpful to know what's allowed. IMO that's worth a bit more friction on the producer side (adding a new member to the enum).

Reply:
I think a big problem with that (not immediately, but in the future) is that we'll break API and ABI every time we want to modify anything in the statistics, furthermore, we would have to register everything in a central place a priori, which is probably not something we really want to do.

@madsbk WDYT?

Reply:
That will avoid the current repetition, but won't help with knowing what the valid keys are.

Reply:
> won't help with knowing what the valid keys are.

But why is it important to know that before it runs?

Reply:
I agree with Peter, being able to add new statistics on the spot can be very useful!